enhancements/teps/0001-option-to-disable-default-injection-of-serviceaccount-rbac.md

```diff
@@ -0,0 +1,124 @@
+authors:
+  - "@savitaashture"
+creation-date: 2021-09-19
+last-updated: 2021-09-19
+status: implementable
+---
+
+# TEP-0001: Option to disable dafault creation of service account and rbac resources on Openshift
+
+<!-- toc -->
+- [Summary](#summary)
+- [Motivation](#motivation)
+  - [Goals](#goals)
+- [Proposal](#proposal)
+  - [User Stories](#user-stories)
+  - [Usage examples](#usage-examples)
+- [Design Details](#design-details)
+  - [Disable creation of RBAC resources at cluster level](#disable-creation-of-rbac-resources-at-cluster-level)
+  - [Scenarios](#scenarios)
+    - [Disable auto creation of RBAC resources](#disable-auto-creation-of-rbac-resources)
+    - [Enable auto creation of RBAC resources (Default behavior)](#enable-auto-creation-of-rbac-resources)
+- [Risk](#risk)
+- [Alternative Approach](#alternative-approach)
+<!-- /toc -->
+
+## Summary
+
+The proposal helps cluster admin to disable auto creation of RBAC resources
+(ServiceAccount, RoleBinding, SCCRoleBinding, CABundlesConfigMap and add Subjects to openshift-pipelines-clusterinterceptors ClusterRoleBinding).
+Related Stories and Epics are
+
+https://issues.redhat.com/browse/SRVKP-1578
+
+https://issues.redhat.com/browse/SRVKP-1670
+
+https://issues.redhat.com/browse/SRVKP-1649
+
+## Motivation
+
+OpenShift Pipelines operator will create a RBAC resources (ServiceAccount(`pipeline`), RoleBinding, SCCRoleBinding, CABundlesConfigMap and Subjects to openshift-pipelines-clusterinterceptors) on all namespace when installed.
+The pipelines-scc-rolebinding has RunAsAny among other things.
+This can be seen as a security issue as some customers have reported. It would be interesting if a cluster-admin could disable this feature on basis of requirement.
+
+### Goals
+
+Provide way to cluster admin to disable auto creation of rbac resources at cluster level. 
+
+### Non-Goals
+
+## Proposal
+
+Installation of OpenShift Pipelines operator by default create RBAC resources on all namespaces.
+cluster admin should have the permission to disable RBAC resource creation at cluster level using `TektonConfig CR`. 
+   
+### User Stories
+As a cluster admin, I want to be able to disable auto creation of ServiceAccount and RBAC resources at cluster level because
+some customers have reported that SCCRolebinding `pipelines-scc-rolebinding` can be seen as a security issue which has **RunAsAny** among other things.
+
+### Usage examples
+
+## Design Details
+
+The main goal of this TEP is to provide ways to cluster admin to disable creation of RBAC resources at cluster level.
+
+### Disable creation of RBAC resources at cluster level
+Cluster admin can create/edit TektonConfig CR and set `disableRbacResourcesForOpenshift` to `true` so that RBAC resources will not create in any of the namespaces in that cluster.
+```yaml
+apiVersion: operator.tekton.dev/v1alpha1
+kind: TektonConfig
+metadata:
+  name: config
+spec:
+  params:
+  - name: disableRbacResourcesForOpenshift
+    value: true
```

Comment:
@vdemeester @nikhil-thomas @sm43 how about this?

Reply:
So a user of the OpenShift platform (and OpenShift Pipelines), will have to set this too, which might be weird (aka why `disableRbacResourcesForOpenShift` instead of just `disableRbacResources` ?).

Because we have this in `params`, and thus not *typed*, any reason for having `ForOpenShift` suffix there ? on Kubernetes setting this would/should be a no-op (at least today).

Reply:
>Because we have this in params, and thus not typed, any reason for having ForOpenShift suffix there ? on Kubernetes setting this would/should be a no-op (at least today).

Yes on Kubernetes setting this leads to no-op
But for clarity i have used `ForOpenShift` suffix

But it make sense to use `CreateRbacResource` 

>So a user of the OpenShift platform (and OpenShift Pipelines), will have to set this too, which might be weird

Ya thats what i assume i mean if user doesn't set that key then how do we decide disabling auto creation :thinking: